src/peft/optimizers/riemannian.py

```diff
@@ -0,0 +1,294 @@
+                            else torch.inverse(scaler.T @ scaler + reg_I)
+                        )
+                        assert scaler.shape[0] == min(p2.data.shape), "wrong dimension"
+                    except RuntimeError:
+                        print("invalid condition")
+                        scaler = None
+
+                    # apply riemannian conditioner
+                    if scaler is not None:
+                        grad = grad @ scaler if grad.shape[1] == scaler.shape[0] else scaler @ grad
+                    # Decay the first and second moment running average coefficient
+                    # In-place operations to update the averages at the same time
+                    exp_avg.mul_(beta1).add_(grad, alpha=(1.0 - beta1))
+                    exp_avg_sq.mul_(beta2).addcmul_(grad, grad, value=1.0 - beta2)
+                    denom = exp_avg_sq.sqrt().add_(group["eps"])
+
+                    step_size = group["lr"]
+                    if group["correct_bias"]:  # No bias correction for Bert
+                        bias_correction1 = 1.0 - beta1 ** state["step"]
+                        bias_correction2 = 1.0 - beta2 ** state["step"]
+                        step_size = step_size * math.sqrt(bias_correction2) / bias_correction1
+
+                    p1.addcdiv_(exp_avg, denom, value=-step_size)
+                    if group["weight_decay"] > 0.0:
+                        p1.add_(p1, alpha=(-group["lr"] * group["weight_decay"]))
+
+                    grad = p2.grad
+                    if grad.is_sparse:
+                        raise RuntimeError(
+                            "Adam does not support sparse gradients, please consider SparseAdam instead"
+                        )
+
+                    state = self.state[p2]
+                    # State initialization
+                    if len(state) == 0:
+                        state["step"] = 0
+                        # Exponential moving average of gradient values
+                        state["exp_avg"] = torch.zeros_like(p2)
+                        # Exponential moving average of squared gradient values
+                        state["exp_avg_sq"] = torch.zeros_like(p2)
+
+                    exp_avg, exp_avg_sq = state["exp_avg"], state["exp_avg_sq"]
+                    beta1, beta2 = group["betas"]
+                    state["step"] += 1
+                    scaler = scaler_temp
+                    try:
+                        reg_I = group["lr"] * torch.eye(min(p1.shape)).to(p1.device)
+                        scaler = (
+                            torch.inverse(scaler @ scaler.T + reg_I)
+                            if p1.shape[0] < p1.shape[1]
+                            else torch.inverse(scaler.T @ scaler + reg_I)
+                        )
+                        assert scaler.shape[0] == min(p1.data.shape), "wrong dimension"
+                    except RuntimeError:
+                        print("invalid condition")
```

Comment:
Could you explain why this is needed? Could we instead check the condition and do something like `if valid_condition: ... else: scaler = None`. Let's completely avoid printing messages.